Feat: add flattenDepth

JaceHensley opened the pull request:
Closes https://github.com/char0n/ramda-adjunct/issues/844

char0n commented:
Thank you for conyributing. I'll provide code review during today evening

char0n commented on src/flattenDepth.js:
```diff
@@ -0,0 +1,36 @@
+  let flatList = list;
+
+  while (currentDept > 0) {
+    flatList = _makeFlat(false)(flatList);
```
this can be extracted and reused
```js
const flatten1 = _makeFlat(false);

/**
 * jsdoc comments....
 */
const flattenDepth = curry((depth, list) => {
  let currentDept = depth;
  let flatList = list;

  while (currentDept > 0) {
    flatList = flatten1(flatList);
    currentDept -= 1;
  }

  return flatList;
}

char0n commented on src/flattenDepth.js:
```diff
@@ -0,0 +1,36 @@
+ * @category List
+ * @sig number -> [a] -> [b]
+ * @param {!number} depth The depth it should be flattened to
+ * @param {!Array} list The provided list
```
@param {!Array} list The array to flatten

char0n commented on src/flattenDepth.js:
```diff
@@ -0,0 +1,36 @@
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.19.0|v2.19.0}
+ * @category List
+ * @sig number -> [a] -> [b]
+ * @param {!number} depth The depth it should be flattened to
```
@param {!number} depth  The maximum recursion depth.

char0n commented on src/flattenDepth.js:
```diff
@@ -0,0 +1,36 @@
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.19.0|v2.19.0}
+ * @category List
+ * @sig number -> [a] -> [b]
```
@sig Number -> [a] -> [b]

Number is a category, that's why the uppercasing

char0n commented on src/flattenDepth.js:
```diff
@@ -0,0 +1,36 @@
+ * @sig number -> [a] -> [b]
+ * @param {!number} depth The depth it should be flattened to
+ * @param {!Array} list The provided list
+ * @return {!Array} The flattened list
```
@return {!Array} Returns the new flattened array

char0n commented on test/flattenDepth.js:
```diff
@@ -0,0 +1,23 @@
+import * as RA from '../src';
+import eq from './shared/eq';
+
```
I'll get to tests Tomorrow. They assertions are fine we just need to fine tune it a little.

Is iterable is a good example how we want to make test recently here: https://github.com/char0n/ramda-adjunct/blob/master/test/isIterable.js

char0n commented:
And again, thank you for you contribution ;]

JaceHensley commented:
@char0n Addressed your feedback, sorry it took me a while to get back to this. Wasn't sure exactly how you wanted me to setup the tests so lmk what I need to change there :)

char0n commented:
Nice job! Thank you for your contribution.

char0n commented:
I continued with the follow-up PR and made tests a little bit more declarative: https://github.com/char0n/ramda-adjunct/pull/958
